src/fmu/dataio/_models/fmu_results/fields.py

```diff
@@ -238,9 +210,8 @@ class Iteration(BaseModel):
     object belongs to.
     """
 
-    id: Optional[int] = Field(default=None)
```

Comment:
No need to be optional anymore. We default this to 0 in the `FmuProvider` if no iteration folder is present.

Reply:
Probably topic for a different discussion, but currently the `id` will default to `0` I guess? But it is only valid for ensembles that are actually iterations (even if we still refer to all ensembles as "iterations").

I.e. if I create an FMU case with 4 prediction ensembles, I will get the same `iteration.id` for all of them. Could be that `iteration.id` should only be populated if the ensemble is actually a prediction.

I believe we shifted all logic from `iteration.id` to `iteration.name` some years ago. Perhaps with the idea of simply discontinuing `iteration.id` completely.